### PR TITLE
refactor update `blas/ext/base/dsumkbn2` to follow current project conventions

### LINKED ISSUE #1506: [RFC]: refactor `blas/ext/base/dsumkbn2` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dsumkbn2`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dsumkbn2) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by liberocks**
Resolves https://github.com/stdlib-js/stdlib/issues/1506.

## Description

> What is the purpose of this pull request?

This pull request refactors the existing implementation to follow the current project convention for `blas/ext/base/dsumkbn2`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves https://github.com/stdlib-js/stdlib/issues/1506
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Review by Planeshifter — APPROVED** [review]
Thanks for this PR; looks all good to me.

File: lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/lib/index.js [hunk 1]
```diff
@@ -39,7 +39,7 @@
 * var dsumkbn2 = require( '@stdlib/blas/ext/base/dsumkbn2' );
 *
 * var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
-* var N = floor( x.length / 2 );
+* var N = 3;;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/lib/index.js:42]: @Planeshifter `floor` should be have been removed. Would be good to ensure that ESLint can run on these examples.

File: lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/lib/ndarray.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ var abs = require( '@stdlib/math/base/special/abs' );
 * var floor = require( '@stdlib/math/base/special/floor' );
 *
 * var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
-* var N = floor( x.length / 2 );
+* var N = 4;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/lib/ndarray.js:50]: Same comment. `floor` should have been removed.

File: lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/lib/ndarray.native.js [hunk 3]
```diff
@@ -40,17 +41,16 @@ var addon = require( './dsumkbn2.native.js' );
 * var floor = require( '@stdlib/math/base/special/floor' );
 *
 * var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
-* var N = floor( x.length / 2 );
+* var N = 4;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/lib/ndarray.native.js:44]: Same comment. `floor` should have been removed.

File: lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/manifest.json [hunk 4]
```diff
@@ -1,40 +1,45 @@
...
-			],
-			"libraries": [
-				"-lm"
-			],
-			"libpath": [],
-			"dependencies": []
-		}
-	]
-}
+  "options": {},
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/manifest.json:2]: @Planeshifter This file is missing build configurations.

File: lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/src/addon.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+  STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+  STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, strideX, argv, 1 );
+	
+  napi_value v;
+  napi_create_double( env, stdlib_strided_dsumkbn2( N, X, strideX ), &v );
+  
+  return v;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsumkbn2/src/addon.c:44]: This file has mixed TAB and space indentation.

**Pull request merged**